### PR TITLE
feat: add kustomization

### PR DESCRIPTION
**Pull request by Azhovan**
**What problem does this PR solve?**:
This PR adds the Kustomization package, which is used to build the Kubernetes kustomizations in application specific testing.

**Which issue(s) does this PR fix?**:
https://d2iq.atlassian.net/browse/D2IQ-98659

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

**Checklist**

- [ ] If the PR adds a version bump, ensure there is no breaking change in Licensing model (or NA).
- [ ] If a chart is changed or app configuration is significantly changed, the chart version is correctly incremented (so that apps are not automatically upgraded from a previous version of DKP).

### CODE REVIEW

File: apptests/kustomize/kustomize.go [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+		resources:   resmap.New(),
+	}
+}
+
+// Build runs the kustomization process on the directory and substitutes the
+// environment variables in the resources.
+func (k *Kustomize) Build() error {
+	for k, v := range k.substitutes {
+		os.Setenv(k, v)
+	}
```
Comment (mhrabovcin) [inline apptests/kustomize/kustomize.go:33]: This function has a side effect of modifying the process environment where it runs, potentially being insecure as it doesn't revert the changes. I think we should add a `defer` call that will revert any possible changes.

Another thing is that this function cannot be called in parallel and so I think it should be lock protected.

Comment (jimmidyson) [reply apptests/kustomize/kustomize.go:33]: Use `envsubst.Eval` instead - do not update the environment like this.

File: apptests/kustomize/kustomize.go [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+		if err != nil {
+			return err
+		}
+
+		// Convert the output string to a resource and append it to the resources map
+		res, err := newResourceFromString(out)
+		if err != nil {
+			return err
+		}
+		k.resources.Append(res)
```
Comment (mhrabovcin) [inline apptests/kustomize/kustomize.go:58]: Should this be reset when `Build` is called? If the `Build` function is called twice there will be the same resources twice.

**Review by mhrabovcin — APPROVED** [review]

**Review by jimmidyson — APPROVED** [review]
